lib/services/user_db_service.dart

```diff
@@ -108,6 +110,87 @@ class UserDbService {
     }
   }
 
+
+
+  // Gets the a random meta task id for a given difficulty for a user.
+  // Returns metaTaskId which can be used to query for the specific task
+  // Returns userIndex which is the index in the user array that can be used to later update remaining tasks for the user
+  Future<Map<String, dynamic>> getRandomMetaTask(
+      String difficulty,
+      ) async {
+    log.infoObj({'method': 'getRandomMetaTask', 'difficulty': difficulty});
+    try {
+      DocumentSnapshot user = await usersCollection.doc(uid).get();
+      MetaTaskDbServices metaTasks = MetaTaskDbServices();
+      try {
+        final random = new Random();
+
+        // If we don't have an array or if it is empty, repopulate the array
+        if (!user.data().toString().contains(difficulty) || user[difficulty].length == 0) {
+          var currentCount = await metaTasks.getCountForDifficulty(difficulty);
+          var list = [for (var i = 1; i <= currentCount['count']; i++) i];
+          usersCollection.doc(uid).update({difficulty: list});
+
+          int index = random.nextInt(currentCount['count']);
+
+
+          log.successObj({
+            'method': 'getRandomMetaTask - success',
+            'metaTaskId': index+1,
+            'userIndex':index,
+          });
+
+          return {"taskId":index+1, "userIndex":index};
+        } else { //array exists in db, so we can pull random task out
+          int len = user[difficulty].length;
+          int index = random.nextInt(len);
+          int metaTaskId = user[difficulty][index];
+          log.successObj({
+            'method': 'getRandomMetaTask - success',
+            'metaTaskId': metaTaskId,
+            'userIndex':index,
+          });
+          return {"taskId":metaTaskId, "userIndex":index};
+        }
+      } on StateError catch(e) {
+        print(e);
```

Comment:
If there is a stateError, should you also return a message to the front-end indicating the error as well?